contracts/contracts/strategies/FraxConvexStrategy.sol

```diff
@@ -0,0 +1,477 @@
+     * 3. locked Frax Staked Convex LP tokens
+     * This assume the locked or unlocked Frax Staked Convex LP tokens equals the Curve LP tokens.
+     * The average is taken prevent the asset balances being manipulated by tilting the Curve pool.
+     *
+     * @dev This will NOT revert if the Frax Convex Staking contract has been shutdown or the
+     * Frax Convex Locking contract has been paused for staking or withdraws.
+     * @param _asset      Address of the asset
+     * @return balance    Total value of the asset in the platform
+     */
+    function checkBalance(address _asset)
+        public
+        view
+        override
+        returns (uint256 balance)
+    {
+        require(_curveSupportedCoin(_asset), "Unsupported asset");
+
+        // Curve LP tokens in this contract. This should generally be nothing as we
+        // should always stake in the Frax Staking contract.
+        uint256 curveLpTokens = IERC20(CURVE_LP_TOKEN).balanceOf(address(this));
+
+        // Get the strategy's balance of Frax Staked Convex LP tokens that are not locked
+        // This will be if there's a pending redeem request.
+        curveLpTokens += IERC20(fraxStaking).balanceOf(address(this));
+
+        // Get the strategy's locked Frax Staked Convex LP tokens
+        curveLpTokens += IFraxConvexLocking(fraxLocking).lockedLiquidityOf(
+            address(this)
+        );
+
+        if (curveLpTokens > 0) {
+            // get_virtual_price is gas intensive, so only call it if we have LP tokens.
+            // Convert the Curve LP tokens controlled by this strategy to a value in USD or ETH
+            uint256 value = (curveLpTokens *
+                ICurvePool(CURVE_POOL).get_virtual_price()) / 1e18;
+
+            // Divide by the number of assets in the Curve pool. eg 2 for the frxETH/WETH pool.
+            // An average is taken to prevent the balances being manipulated by tilting the Curve pool.
+            // No matter what the balance of the asset in the Curve pool is, the value of each asset will
+            // be the average of the Curve pool's total value.
+            // This assumes all assets in the Curve pool have the same number of decimals.
+            balance = value / CURVE_POOL_ASSETS_COUNT;
+        }
+    }
+
+    /**
+     * @dev Collect accumulated CRV, CVX and FXS rewards and send to Harvester.
+     * The Frax Convex Locking contract can be paused for collecting rewards. If so,
+     * `collectRewardTokens` will fail with "Rewards collection paused".
+     * This will revert if there is not enough FXS rewards in the
+     * Frax Convex Locking contract. See dev note in contract Natspec for more details.
+     */
+    function collectRewardTokens()
+        external
+        override
+        onlyHarvester
+        nonReentrant
+    {
+        // Collect CRV, CVX and FXS rewards from staking contract
+        IFraxConvexStaking(fraxStaking).getReward(address(this));
```

Comment:
The `getReward` function can be invoked using the harvester address, allowing for a reduction in one transfer for both the `fraxStaking` and `fraxLocking` contracts.

Reply:
interesting, so Frax allow rewards to be delegated. In that case, the strategy can delegate rewards to the Harvester.

This requires a change to the Harvester which would be out of scope for this deployment. I'll add this Harvester feature request to the backlog